Weight proofs new version

almogdepaz opened the pull request:
Smaller proof format

Sampled segments change every sub epoch and not every peak

Include proofs only in sampled segments

Added Wp capabilities in handshake to support both versions
so node will try using new version if peer is capable old version if not

suggestions about a better naming conventions then _v2 are appreciated

almogdepaz committed:
test

almogdepaz committed:
weight proof v2

almogdepaz committed:
node integrations

almogdepaz committed:
lint

almogdepaz committed:
log

almogdepaz committed:
startup segments

almogdepaz committed:
fix name

almogdepaz committed:
fix var

almogdepaz committed:
minor fixes

almogdepaz committed:
logs

almogdepaz committed:
lint

almogdepaz committed:
move capabilities to server

almogdepaz committed:
test backwards compatibility

almogdepaz committed:
fix test vars

almogdepaz committed:
fix test vars

almogdepaz committed:
Merge branch 'main' of github.com:Chia-Network/chia-blockchain into main

almogdepaz committed:
Merge branch 'main' of github.com:Chia-Network/chia-blockchain into main

almogdepaz committed:
Merge branch 'main' of github.com:Chia-Network/chia-blockchain into main

almogdepaz committed:
skip tx filter in v2

almogdepaz committed:
lint

Nikolaj-K commented on chia/consensus/blockchain.py:
```diff
@@ -730,6 +730,22 @@ async def get_sub_epoch_challenge_segments(
+        segments: Optional[
+            List[SubEpochChallengeSegmentV2]
+        ] = await self.block_store.get_sub_epoch_challenge_segments_v2(ses_block_hash)
+        if segments is None:
+            return None
+        return segments
```
lines 745+746 are redundant

almogdepaz replied:
fixed, Thanks @Nikolaj-K

almogdepaz committed:
wallet handles v2

almogdepaz committed:
clean code

almogdepaz committed:
test wallet integration

almogdepaz committed:
lint

almogdepaz committed:
remove comment

almogdepaz committed:
Merge branch 'main' into weight_proofs_new_version

# Conflicts:
#	chia/full_node/full_node.py
#	chia/server/server.py
#	tests/core/full_node/full_sync/test_full_sync.py

almogdepaz committed:
lint

almogdepaz committed:
Merge branch 'main' into weight_proofs_new_version

# Conflicts:
#	chia/protocols/protocol_message_types.py
#	chia/wallet/wallet_node.py
#	chia/wallet/wallet_state_manager.py
#	tests/core/full_node/full_sync/test_full_sync.py
#	tests/wallet/sync/test_wallet_sync.py

almogdepaz committed:
change test config

almogdepaz committed:
change test config

mariano54 committed:
Remove debug logging (#7061)

mariano54 committed:
farmer: Reload keys when config changes, so we can detect new plot NFTs (#7060)

almogdepaz committed:
fix import

arvidn committed:
lower the RAM threshold to 800 MB (the largest test use 144 MB right now). Also improve printing of results (#6972)

emlowe committed:
New RPC to check keys prior to deleting them (#6894)

* New RPC to check keys prior to deleting them

* Updates based on PR comments

* Make sure to stop wallet before start again

cmmarslender committed:
M1 Installer (#7092)

nirajpathak13 committed:
Split linux installers (#6987)

Move .rpm and .deb installers into individual workflows

arvidn committed:
in strict/safe mode, require integers to use the minimal/canonical encoding (#7071)

emlowe committed:
Improves error handling for unknown protocol types (#7073)

* Improves error handling for unknown protocol types

* Add test for invalid protocol

mariano54 committed:
Fix DB invalid format for additional_coin_spends

nirajpathak13 committed:
rename build script to match corresponding yml file

nirajpathak13 committed:
update arm64 builder

almogdepaz committed:
fix v2 tests

almogdepaz committed:
factor out get_iters_and_input

almogdepaz committed:
remove duplicate code

almogdepaz committed:
dont send full icc vdf info

almogdepaz committed:
lint

sayomaki committed:
Minor fix to wrong error message printed

cmmarslender committed:
Split secret check to signing and aws secrets, so we can still do AWS actions when we dont have a signing cert available (#7142)

wjblanke committed:
updated gui to latest

aqk committed:
Enforce HTTPS when fetching pool server configs for mainnet (#7148)

* Enforce HTTPS when fetching pool server configs for mainnet

* Update chia/cmds/plotnft_funcs.py

Co-authored-by: dustinface <35775977+xdustinface@users.noreply.github.com>

* Update chia/cmds/plotnft_funcs.py

Co-authored-by: dustinface <35775977+xdustinface@users.noreply.github.com>

* Update chia/farmer/farmer.py

Co-authored-by: dustinface <35775977+xdustinface@users.noreply.github.com>

Co-authored-by: dustinface <35775977+xdustinface@users.noreply.github.com>

xdustinface committed:
farmer: Always check the pool URL for `https://` (#7152)

xdustinface committed:
farmer|cmds: Improve remote harvester info (#6979)

* farmer|cmds: List harvester by IP/peer_node_id instead of IP/port

* farmer: Cache harvester responses

* test: Trigger `update_cached_plots` and adjust `get_plots` format

* farmer|cmds|test: Adjust RPC output of `get_plots`

* rename: Farmer RPC `get_plots` to `get_harvesters` + rename related code

* farmer: Fix remove loop indentation

* farmer: Catch exceptions in `_periodically_clear_cache_and_refresh_task`

* cmds: Distinguish between local and remote harvesters

* cmds: Explicitly check against 0

* cmds: Rename `print_harvesters` -> `process_harvesters`

altendky committed:
Correct 'leave' command --help

xdustinface committed:
farmer: Update `payout_instructions` on the pool if changed (#7161)

* farmer: Refactor `update_pool_farmer_info`

Return the actual `GetFarmerResponse` object and a flag if the farmer is
known on the pool or not.

* farmer: Call `PUT /farmer` if the `payout_instructions` changed

* farmer: Force farmer update, don't directly call `update_pool_state`

Let the periodical task handle the update.

* farmer: Temporary fix for `Streamable` not supporting JSON `Optional`

* tests: Call `update_pool_state` after `set_payout_instructions`

wjblanke committed:
updated gui

aqk committed:
longer system test timeout (#7172)

altendky committed:
make instructions in the issue template be comments (#7177)

This avoids the instructions being shown in the final PR submission.

hoffmang9 committed:
Update GUI submodule for plot count fix (#7180)

hoffmang9 committed:
Changelog for 1.2.0 release - draft (#7145)

* Changelog for 1.2.0 release - draft

* Thanks Atom! - @gldecurtins corrected.

* fixed some typos

* Remove accidental cruft

* "Updated chiapos"

Co-authored-by: Adam Kelly <338792+aqk@users.noreply.github.com>

* More precision about pre-release plot NFT, OG plots.

Co-authored-by: Adam Kelly <338792+aqk@users.noreply.github.com>

* Hopefully capturing @efishcent's changes

* Changelog updates

* More details on RPC and other changes

* Nits and a date

* Final two GUI enhancements

* Markdown lint

* remote harvester issue list

* Let GH link the commit SHAs

* Note re wallet resync

Co-authored-by: Adam Kelly <338792+aqk@users.noreply.github.com>

Co-authored-by: Roy Natian <roynatian@gmail.com>
Co-authored-by: Adam Kelly <338792+aqk@users.noreply.github.com>
Co-authored-by: Mariano <sorgente711@gmail.com>

cmmarslender committed:
Fix folder for m1 torrents (#7186)

arvidn committed:
fix duplicate-output test to not also do a double-spend (#7176)

sargonas committed:
GH Action to tag and close stale issues (#7140)

* Create stale-issue.yml

Created initial Stale Issue/PR script config with the following configuration parameters:

Run the script every day at 4am PST (11am UTC)
Flag issues as stale after 14 days of no activity with stale-issue label.
Flag PRs as stale after 60 days of no activity with stale-pr label.
Close stale issues after a further 7 days of no activty.
Do NOT close stale PRs, only warn with comments.
Ignore stale status for any Issues or PRs with an assigned milestone, or an assignee.
Remove stable labels if they have been updated

This change will not be put into a PR for review until this entire flow has been tested on release-test first.

* Update stale-issue.yml

changing to main to incorporate some needed fixes

Quexington committed:
Fix load.clvm

almogdepaz committed:
move peak look to _start (#7270)

xdustinface committed:
farmer|harvester|server: Fix harvester cache updates, bump message size, increase timeout (#7244)

* farmer: Increase timeout for `HarvesterAPI.request_plots` requests

Set it to `UPDATE_HARVESTER_CACHE_INTERVAL`.

* farmer|harvester: Add some debug logs around `get_harvester` stuff

* server: Bump size limit for `respond_plots` from 10MB/msg to 100MB/msg

Those messages can become pretty big now for large farms.

* farmer: Keep also track of failed cache update attempts

Prior to this commit the farmer called the `request_plots` every second 
for each harvester as long as they failed to respond properly. Since the 
rate limit was 10/minute this lead to hitting the rate limit if the 
harvester didn't responds for 10 tries in a row for whatever reason.

This commit changes the behaviour to always keep track of request 
attempts even if they end up in a timeout to really only re-try every 
60s no matter what.

* farmer: Introduce and use `HarvesterCacheEntry`

feldsam committed:
Added support for Fedora in install-gui script (#5957)

* Support Fedora in install-gui script

* Update install-gui.sh

* Update install-gui.sh

cmmarslender committed:
Fix m1 version (#7321)

* Rename job - not catalina

* Cleanup leftovers from previous runs. #7309

paulhainsworth-chia committed:
Changelog for 1.2.1 (#7422)

richardkiss committed:
Rename coin solution to coin spend (#6841)

* Rename `CoinSolution` as `CoinSpend` everywhere.

* Run `black` everywhere.

beechord committed:
coordinate blueboxes better (#7175)

* bluebox: simplify handling of uncompact orphan blocks

* bluebox: fix redundant work done by blueboxes at the bottom of the chain

* bluebox: do not tarpit when the chain is nearly fully compact

Co-authored-by: beechord <@>

cmmarslender committed:
Pin pyinstaller to a known working version for m1 (#7432)

lukemcgregor committed:
[DOCS] Updated main readme to point to pooling guide (#7280)

sayomaki committed:
Fix wrong typings for get_puzzle_and_solution (#7326)

Commit:
fix error when sync with height 0 (#7419)

hoffmang9 committed:
Revert f4a685

dependabot[bot] committed:
Bump actions/setup-node from 2.1.5 to 2.2.0 (#7144)

Bumps [actions/setup-node](https://github.com/actions/setup-node) from 2.1.5 to 2.2.0.
- [Release notes](https://github.com/actions/setup-node/releases)
- [Commits](https://github.com/actions/setup-node/compare/v2.1.5...v2.2.0)

---
updated-dependencies:
- dependency-name: actions/setup-node
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Quexington committed:
Added get_coin_records_by_parent_id(s) (#7310)

* Added new RPCs to get coins by their parent

* Got rid of the individual request in favor of requiring a list of puzzle hashes

ChiaMineJP committed:
Fixed type annotation (#7445)

aqk committed:
Add resource usage check to generated template. Add warning to generated workflows. (#7181)

* Add resource usage check to generated template. Add warning to generated workflows

* Revert submodule commit

cmmarslender committed:
Update mozilla-ca (#7457)

Quexington committed:
Lightweight spend simulator (#7256)

* added a new lightweight simulator and updated a single test to use it

* Implemented full node rpc methods on the node (untested)

* moved locations and separated out the node client

* Lint fix

* Added barebones testing

* python black fixes

* Moved to RAM CoinStore

* Renamed Node to SpendSim and NodeClient to SimClient

* CoinSolution is CoinSpend now

justinengland committed:
Ca cert automation (#7472)

* removing changes that are not mine

* fixing new lines?

* Update mozilla-ca-cert.yml

* Update mozilla-ca-cert.yml

* trying different formatting

* fixing bad spacing ln 32

* Format

* More line endings?

Co-authored-by: Chris Marslender <chrismarslender@gmail.com>

arvidn committed:
add test for minting coin check in block_body_validation (#7450)

emlowe committed:
Use bundled CA store for pool https (#7461)

seraphik committed:
Minor bug fix : dnf requires elevated priveleges to enable modules. (#7493)

Tested on Fedora and Rocky.

Co-authored-by: h <h@localhost.localdomain>

cmmarslender committed:
Swap the start release notify to the dedicated notify runner (#7507)

dependabot[bot] committed:
Bump github/super-linter from 4.2.2 to 4.3.0 (#7460)

Bumps [github/super-linter](https://github.com/github/super-linter) from 4.2.2 to 4.3.0.
- [Release notes](https://github.com/github/super-linter/releases)
- [Commits](https://github.com/github/super-linter/compare/v4.2.2...v4.3.0)

---
updated-dependencies:
- dependency-name: github/super-linter
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

cmmarslender committed:
Add cleanup script to the beginning of all workflows to ensure a clean slate (#7515)

mariano54 committed:
Ms.pool wallet spam2 (#7447)

* pool_wallet: Return correct balance, ignoring spam

* pool_wallet: Raise error if there are no rewards to claim. Adds test

almogdepaz committed:
fix err type in wallet_blockchain.py (#7527)

dependabot[bot] committed:
Bump blspy from 1.0.2 to 1.0.3 (#7462)

Bumps [blspy](https://github.com/Chia-Network/bls-signatures) from 1.0.2 to 1.0.3

xdustinface committed:
chia|tests: Add `node_type` parameter to `get_connections` RPC (#7492)

* rpc|server: Add Optional `node_type` paramter to `get_connections`

* farmer: Query `HARVESTER` connections only

* tests: Basic test for `node_type` parameter

mariano54 committed:
cli: Show plot count in `chia plotnft show` (#7481)

* cli: Show plot count in `chia plotnft show`

* cli: Ignore non-pooled plots

arvidn committed:
remove left-over debug logging from test_wallet_pool_store (#7533)

mariano54 committed:
cli: Use correct type annotations (#7482)

* cli: Use correct type annotations

* cli: Respond to comments

* cli: Add more missing Optionals

aqk committed:
Warn user not to send money to Pay to Singleton ("pool contract") address (#7541)

arvidn committed:
add test for ephemeral coins with relative time/height locks (#7502)

* add test for ephemeral coins with relative time/height locks

* additional checks

* formatting

* rename parameter

* don't include condition in first spend

* move test into a better test class

* fixes

* formatting

* simplify test

* make ephemeral coin test not depend on wall clock. Make the initial transaction freeze test also not depend on wall clock. Add tests for ASSERT_SECONDS_ABSOLUTE

* formatting

Quexington committed:
Added an option to set the default constants on the simulator (#7495)

arvidn committed:
fix test_singleton (#7559)

arvidn committed:
convert test_rom.py to use pytest (#7560)

arvidn committed:
add mempool tests (#7519)

* add mempool tests

* Update tests/core/full_node/test_mempool.py

Co-authored-by: Richard Kiss <him@richardkiss.com>

* Update tests/core/full_node/test_mempool.py

Co-authored-by: Adam Kelly <338792+aqk@users.noreply.github.com>

* Update tests/core/full_node/test_mempool.py

Co-authored-by: Richard Kiss <him@richardkiss.com>

* Update tests/core/full_node/test_mempool.py

Co-authored-by: Adam Kelly <338792+aqk@users.noreply.github.com>

Co-authored-by: Richard Kiss <him@richardkiss.com>
Co-authored-by: Adam Kelly <338792+aqk@users.noreply.github.com>

Quexington committed:
Added backwards compatibility for coin_solutions in push_tx (#7574)

Quexington committed:
Include all chialisp files in source distribution (#7571)

cmmarslender committed:
CentOS runner for rpm builds (#7552)

* disable build links

* Disable binary stripping - #3846

Causes error:
`ImportError: /usr/lib/chia-blockchain/resources/app.asar.unpacked/daemon/chiavdf.cpython-39-x86_64-linux-gnu.so: ELF load command address/offset not properly aligned`

* Quote NODE_ROOT

* Add workaround for building on centos

* Remove the cache stages, since this is on self-hosted now and we remove it all anyways

wjblanke committed:
update blspy to 1.0.4 (sans GIL release)

mishan committed:
Display successful points in `chia plotnft show` (#7611)

The GUI has a useful "Percent Successful Points" display and it just gets this
from `get_pool_state`

Since `chia plotnft show` is already calling this API and getting this
data, I've updated the output to show it

felixbrucker committed:
Fix invalid content-type header in pool api requests (#7589)

xdustinface committed:
cmds|util|tests: Drop `0x` prefix from JSON conversions (#7645)

Not sure if that has been done for a specific reason but i feel like its 
better to not have them in general. Let me know if there are any 
objections, im happy to close this PR if its for a reason i missed.

xdustinface committed:
farmer: Use `hexstr_to_bytes` in `set_payout_instructions` (#7646)

This is to also allow hex strings prefixed with `0x` here which is 
required because we prefix the strings in JSON conversion here 
https://github.com/Chia-Network/chia-blockchain/blob/4fbc2919692260ed15604dff28e1d50cf9845375/chia/util/json_util.py#L20-L23

wjblanke committed:
updated gui to 96b8def9dadf0bd6af0b5c66aee7a2abe473a2ce

dependabot[bot] committed:
Bump github/super-linter from 4.3.0 to 4.5.0 (#7675)

Bumps [github/super-linter](https://github.com/github/super-linter) from 4.3.0 to 4.5.0.
- [Release notes](https://github.com/github/super-linter/releases)
- [Commits](https://github.com/github/super-linter/compare/v4.3.0...v4.5.0)

---
updated-dependencies:
- dependency-name: github/super-linter
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump actions/setup-node from 2.2.0 to 2.3.0 (#7674)

Bumps [actions/setup-node](https://github.com/actions/setup-node) from 2.2.0 to 2.3.0.
- [Release notes](https://github.com/actions/setup-node/releases)
- [Commits](https://github.com/actions/setup-node/compare/v2.2.0...v2.3.0)

---
updated-dependencies:
- dependency-name: actions/setup-node
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Quexington committed:
Partial reversion of CoinSpend rename for API compatibility (#7636)

* Partial reversion of the coin spend rename

* Make `SpendBundle` json-serialize the classic way by default.

Co-authored-by: Richard Kiss <him@richardkiss.com>

emlowe committed:
Option to revert back to sequential read (#7644)

* Option to revert back to sequential read

* reverse logic for disable

* Add parallel_read to initial-config

wjblanke committed:
upped bls to 1.0.5

wjblanke committed:
update gui to bd3f816ef60402717a0c8636888da94abf342b46

wjblanke committed:
reverting 7645 (#7708)

emlowe committed:
More friendly error reporting for peername errors (#7716)

chadwick2143 committed:
Add set_harvester_port in chia.cmds.configure (#6937)

aarcro committed:
Add timing metrics to plot check (#5109)

* Add timing metrics to plot check

* Remove averge time, add quality check time

Formatting fixed

mariano54 committed:
Ms.remove proof limit (#7745)

* Remove proof limit for mainnet, but not testnet

* farmer: Revert typo

mariano54 committed:
Ms.refresh plots (#7744)

* farmer: Send update to GUI when plots or harvesters change

* farmer: Increase cache interval to help large farmers

mgraczyk committed:
Fix use of print_exc in farmer (#7754)

print_exc prints to a file, while format_exc returns a string:

https://docs.python.org/3/library/traceback.html#traceback.format_exc

nirajpathak13 committed:
rename windows checksum to be consistent with other installer checksums (#7756)

xdustinface committed:
farmer|tests: Always bump `last_update` in `update_cached_harvesters` (#7792)

* farmer: Always bump `last_update` in `update_cached_harvesters`

* tests: Wait a bit for the harvester in `test_farmer_harvester_rpc.py`

Playwo committed:
Case insensitive payout_instructions comparison (#7768)

paulhainsworth-chia committed:
changelog for 1.2.3 (#7702)

* changelog for 1.2.3

* Changelog for 1.2.3 updated 7-26-21

arvidn committed:
extend initial freeze test (#7580)

dependabot[bot] committed:
Bump github/super-linter from 4.5.0 to 4.5.1 (#7824)

Bumps [github/super-linter](https://github.com/github/super-linter) from 4.5.0 to 4.5.1.
- [Release notes](https://github.com/github/super-linter/releases)
- [Commits](https://github.com/github/super-linter/compare/v4.5.0...v4.5.1)

---
updated-dependencies:
- dependency-name: github/super-linter
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

almogdepaz committed:
add block height assertions after block farming (#7526)

add tx confirmed assertions
use time_out_assert only when needed

rostislav committed:
Add pairing cache for faster aggregate signature verification (#7484)

* Add pairing cache to speed up aggregate signature verification

* Reduce excessive logging in test_performance.py

* Fix issues in cached_bls found in code review

Co-authored-by: Rostislav <rostislav@users.noreply.github.com>

darkverbito committed:
Fixed an issue on initial creation of a coloured coin where code always falls into default else clause due to lack of type conversion (#7851)

emlowe committed:
EL.Set version in package.json before packaging (#7864)

* Set version in package.json before packaging

cmmarslender committed:
remove duplicated sed command in rpm build (#7865)

cmmarslender committed:
Build RPM in Chia Centos7 base image instead of VM (#7866)

mgraczyk committed:
Fix type annotation in sync_store (#7911)

aqk committed:
Disable DID wallet tests while DID wallet is under construction. (#7862)

xdustinface committed:
refactor: Rename folder `tests/core/types` to `tests/core/custom_types` (#7852)

* refactor: Rename folder `tests/core/types` to `tests/core/custom_types`

Seems like the name `types` in the folder conflicts there with some python
standard stuff because i just can't get the debugger in PyCharm to work
without this renaming. Instantly getting the following error:

```
ImportError: cannot import name 'MappingProxyType' from 'types'
(/home/me/chia/chia-blockchain/tests/core/types/__init__.py)
Traceback (most recent call last):
  File
"/snap/pycharm-professional/248/plugins/python/helpers/pydev/pydevd.py",
line 8, in <module>
    from contextlib import contextmanager
  File "/usr/lib/python3.8/contextlib.py", line 7, in <module>
    from types import MethodType
ImportError: cannot import name 'MethodType' from 'types'
(/home/me/chia/chia-blockchain/tests/core/types/__init__.py)
```

If someone has an idea how else to fix this please tell me!

* github: Run `build-workflows.py`

With DID tests dropped!

xdustinface committed:
pre-commit: Add initial config and run it for all files (#7053)

* pre-commit: Add initial `.pre-commit-config.yaml`

* pre-commit: Apply hook `end-of-file-fixer`

* pre-commit: Apply hook `trailing-whitespace`

* pre-commit: Apply hook `debug-statements`

* contrib: Add `pre-commit` section

cmmarslender committed:
Pyinstaller 4.5 (#7915)

* Pyinstaller 4.5 for M1

* Update linux builds to 4.5

* Update windows to pyinstaller 4.5

* Pyinstaller 4.5 for intel mac

arvidn committed:
remove unnecessary indentation in block_body_validation (#7916)

arvidn committed:
don't save unknown conditions (#7548)

* don't save unknown conditions

* remove UNKNOWN enum from ConditionOpcode enum

cmmarslender committed:
Swap to the chia version of notarize CLI that retries status and has exit codes when it fails (#7919)

arvidn committed:
remove trailing whitespace in generated workflow files (#7921)

Playwo committed:
Remove index on coin_record spent column (#7794)

* Remove index on spent column

* Ignore index for queries on spent column

* Use IndexBy to select index to use

* Remove IndexedBy from coin_parent query

* Readd index on coin_spent

* Remove whitespaces

Co-authored-by: dustinface <35775977+xdustinface@users.noreply.github.com>

* Split query into serveral lines

Co-authored-by: dustinface <35775977+xdustinface@users.noreply.github.com>

* Split query into serveral lines

Co-authored-by: dustinface <35775977+xdustinface@users.noreply.github.com>

* Add missing newline

* Fix lint/black issue

Co-authored-by: dustinface <35775977+xdustinface@users.noreply.github.com>
Co-authored-by: Earle Lowe <e.lowe@chia.net>

aqk committed:
Separate locking and non-locking cases for get_confirmed_balance_for_wallet to allow calling a few wallet_state_manager methods while already under the wallet_state_manager lock (#7790)

arvidn committed:
remove initial transaction freeze feature (#7922)

dependabot[bot] committed:
Bump styfle/cancel-workflow-action from 0.9.0 to 0.9.1 (#7935)

Bumps [styfle/cancel-workflow-action](https://github.com/styfle/cancel-workflow-action) from 0.9.0 to 0.9.1.
- [Release notes](https://github.com/styfle/cancel-workflow-action/releases)
- [Commits](https://github.com/styfle/cancel-workflow-action/compare/0.9.0...0.9.1)

---
updated-dependencies:
- dependency-name: styfle/cancel-workflow-action
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

cmmarslender committed:
Create keychain for use in CI jobs (#7933)

* Create keychain for use in CI jobs

* Add local keychain in the template for workflow generation

Quexington committed:
Added an option to sign bytes as well as UTF-8 strings (#7923)

cmmarslender committed:
Swap password to just be "foo" (#7946)

paninaro committed:
Keyring passphrase protection (#7249)

* Moved keyring handling into a KeyringWrapper class

* Update click to 8.0.x for prompt_required support

* Renamed KeyringWrapper to _KeyringWrapper

* Expose password management options on Linux

* CLI support for setting/removing a password

* Global option for specifying the master password

* Cache the password instead of setting on the context

* Password bootstrapping during chia init

* Tidying up _KeyringWraper's interface

* Initial pass migrating the legacy keyring contents

* Encryption/decryption of keyring.yaml contents

* FileKeyring backend encrypts with ChaCha20Poly1305

* Tightened up keyring migration and initialization

* Fixed issues identified by linters

* Remove root_path from Keychain

* Prevent double-migration if setting master passwd

* KeyringWrapper tests are mostly complete

* FileKeyring will now honor the service param

* Tests for get/set/delete password

* Formatting/commenting updates

* Writer lock support with tests - WIP

* keyring.yaml is now watched for modifications

* Reader/Writer lock for get/delete password

* Fixed linter issues

* Reader lock tests

* Formatting update

* Hook up CHIA_ROOT support for KeychainWrapper

* Quick fix to address test failures

* Fixed failures when existing legacy keyring exists

* Fixed test failures caused by reusing the same temp dir

* keyring.yaml now lives in ~/.chia_keys by default. Can be overridden with CHIA_KEYS_ROOT or --keys-root-path

* Fixed migration failure when setting a password (not using the default)

* KeyringWrapper now uses supports_keyring_password to determine if a FileKeyring should be used. Patched tests to work regardless of whether supports_keyring_password return False

* The daemon now takes a --have-gui option that will prevent calling check_keys() during startup. If the keyring is locked, we want the GUI to prompt for the password.

* Added is_keyring_locked RPC call

* Added 'unlock_keyring' RPC command

* Added KeychainProxy and KeychainServer to handle RPC messages related to keyring operations. WalletNode no longer directly accesses the Keychain class.

* Turn on macOS support for testing keyring passwords

* Fixed get_key_for_fingerprint to use the ocal keychain if the platform doesn't need to remotely access the daemon's keychain.

Fixed key reconstruction when sent over RPC.

* Farmer now accesses the keychain over RPC

* Fixes for linter issues and some restructuring to support tests that use setup_nodes.py

* Couple of fixes to unblock the GUI from launching when a keyring password is set

* Added a keychain RPC call for add_private_key()

* Added remaining keychain proxy RPC calls for delete_key_by_fingerprint and delete_all_keys

* Check for None when inspecting request arguments

* Run check_keys after unlocking the keyring when the daemon is launched via GUI

* Added check_keys RPC method.
Fixed deserialization of key entropy in get_all_private_keys. This was preventing the GUI from being able to show key details.

* Added get_first_private_key to keychain_server/proxy.
create_plots now uses the keychain proxy when launched from the daemon.

* Added a comment about KeychainProxy in chia plots check

* Workaround import conflict  when importing from 'tests.*' due to fasteners name conflict

* Simulator now uses KeychainProxy if launched by the daemon.
KeychainServer/Proxy now takes keychain user/testing params for testing scenarios.

* Added "set_keyring_passphrase" RPC message

* Reworking KeychainProxy usage to handle local keychain tests and RPC keychain tests.

* Replace my prior usage of asyncio.run() with asyncio.get_event_loop().run_until_complete()

* Silencing file_keyring logging for the moment.

* Updated tests to use test keychains and appropriate BlockTools construction

BlockTools should now be created with create_block_tools(_async) to handle async scenarios.

Updated block_tools to be async compatible

Updated fasteners to fix installation of top-level 'tests' in site-packages

* Added 'remove_keyring_passphrase' RPC message to the daemon

Minor tweak to TempKeyring to default to some test params

* Fixed linter issues

* Remove flake8 ignore statement now that the fasteners module has been updated

* Some initial renaming changes: password -> passphrase

* Fixed wallet RPC issue where get_key_for_fingerprint wasn't awaited-upon.

Fixed legacy keyring initialization (for migration scenarios)

* Fixed improperly merged file

* Fixed linter issues.
More renaming.

* Updated spots that were still using an incorrect keychain call

* Renamed use_password_cache, obtain_current_password

* Renamed supports_keyring_password

* Renamed has_master_password

* Renamed has_cached_password, get_cached_master_password

* Linter fixes

* Renamed master_password_is_valid

* Renamed set_cached_master_password

* Renamed set_master_password

* Renamed remove_master_password

* Renamed has_cached_master_password

* Renaming in file_keyring and keyring_wrapper

Updated default keyring payload used for tests

* Renamed get_password
Other renaming updates

* Renamed set_password
Other renaming updates

* Renamed remaining password occurrences (where appropriate)

* password -> passphrase

* Added tests for setting an emoji and Japanese master passphrase

* Attempt to notify the daemon when a keyring passphrase is set/updated/removed

* Missed one password -> passphrase replacement.

* Fixed some file synchronization issues found when running tests on macOS

* Adjusted timeout values for test_writer_lock_reacquisition_failure for macOS.

* Removed logging statements previously added for debugging

* Prompt for keyring passphrase up-front when launching a service.

Changed --have-gui flag to --wait-for-unlock

* Updated set_keyring_passphrase RPC message to fix optional current_passphrase param when the keyring is using the default passphrase.

* Minor test cleanup to deduplicate some code.

* Fixed regression when setting a new master passphrase

* Minor refactoring and docs/commenting updates

* Renaming password -> passphrase went too far. Keyring backends use password terminology for compatibility with third party backends.

* Disabling macOS support (previously added for testing only)

* Disabling passphrase support in preparation for sending out the PR

* Fixed improper merge (vscode didn't save changes during rebase)

* Update chia/cmds/init_funcs.py

Co-authored-by: Adam Kelly <338792+aqk@users.noreply.github.com>

* skip_check_keys -> should_check_keys

* Shuffling some imports around to break cycles reported by LGTM

* Handle unlocking the daemon if it's already launched and waiting for unlock.

* Replaced uses_keychain_proxy decorator in farmer.py. Fixed async usage of get_reward_targets.

Linter/reformatting fixes

* Replaced uses_keychain_proxy decorator with a clearer method.

* Cleanup the temp keyring dir using shutil.rmtree()

* Restored self._root_path (had been changed to self.root_path)

* Minor cleanup

* ensure_keychain_proxy() now throws if connect_to_keychain_and_validate() fails

* Plot key resolution now yields a PlotKeys object which can be passed into create_plots.

De-indented test_invalid_icc_sub_slot_vdf to keep git blame tidy.

* Added 'keyring_status' daemon RPC message to support the GUI

* Minor changes relating to PR feedback

* Addressed more PR feedback (mostly type annotations)

* Commented-out macOS file keyring usage. This can be re-enabled for testing purposes.

* Addressed test failures that require multiple keyrings in the same process. Each TempKeyring will now set a custom KeyringWrapper instance.

* Fixed logic for communicating user_passphrase_is_set in the keyring_status RPC response.

* Updated type annotations and method signature for set_passphrase to expect a string instead of bytes.

* Fixed Wallet RPC tests

* Fixed full_node_store tests. BlockTools should be created using the create_block_tools(_async) function(s)

* Fixed test failures in test_pool_rpc

* Fixed test_daemon. After BlockTools.setup_plots is run, the config file needs to be re-read to refresh stale plot_directories.

* Suppressing LGTM false positives regarding passphrase leakage in CLI error output. Seems that LGTM sees MIN_PASSPHRASE_LEN as sensitive data.

* Second attempt at suppressing LGTM false positives

* Third attempt at addressing LGTM false positives

* Removed test_keyring_wrapper param from Keychain ctor. Test setup now sets the keyring_wrapper property directly.

* Reformatting

* More targeted update of the test config to refresh just the "plot_directories" value

* More LGTM suppressions

Co-authored-by: Adam Kelly <338792+aqk@users.noreply.github.com>
Co-authored-by: wjblanke <wjb98672@gmail.com>

dependabot[bot] committed:
Bump bitstring from 3.1.7 to 3.1.9 (#7676)

Bumps [bitstring](https://github.com/scott-griffiths/bitstring) from 3.1.7 to 3.1.9.
- [Release notes](https://github.com/scott-griffiths/bitstring/releases)
- [Changelog](https://github.com/scott-griffiths/bitstring/blob/master/release_notes.txt)
- [Commits](https://github.com/scott-griffiths/bitstring/commits)

---
updated-dependencies:
- dependency-name: bitstring
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

almogdepaz committed:
Merge remote-tracking branch 'origin/weight_proofs_new_version' into weight_proofs_new_version

paninaro committed:
When proxying to a local keychain, don't attempt to communicate over the websocket. (#7949)

xdustinface committed:
refactor: Introduce `write_uint32` and `parse_uint32` in `streamable.py` (#7932)

* refactor: Introduce `parse_uint32` in `streamable.py`

* refactor: Introduce `write_uint32` in `streamable.py`

* tests: Add `test_uint32` in `test_streamable.py`

Adds tests for `write_uint32` and `parse_uint32`

xdustinface committed:
chia|tests: Introduce `PlotManager` + some plot loading improvements and fixes  (#7848)

* harvester|plotting|tests: Introduce `PlotManager` class

This moves all plots related and plot-directory related stuff from the
harvester into the class `PlotManager`, adjusts all related code
accordingly and adds some extra wrappers there.

* harvester|plotting|tests: Return how many new plots were loaded

* plotting: Fix `failed_to_open_filenames` re-try interval

With `< 1200` it just tries it on the next refresh.

* plotting: Fix and improve duplicates handling

* harvester|plotting: Thread locks for `PlotManager.plots`

* chia|tests: Load plots in batches

* chia|tests: Move plot refreshing into a separate thread

* plotting: Properly handle removed plots

And fix tests accordingly. It seems like this fix https://github.com/Chia-Network/chia-blockchain/pull/3350 wasn't really a fix, rather adjusting to allow for a bug?

* plotting|harvester|tests: Introduce `PlotRefreshResult`

* tests: Expand `test_farmer_harvester_rpc.py`

* chia|tests: Move some stuff from `plot_tools.py` into new file `util.py`

* refactor: Rename `plot_tools.py` to `manager.py`

* chia|tests: Use pure dataclass for `PlotsRefreshParameter`

With `uint16` as type saving to config doesn't work, this is a
preparation for the next commit.

* harvester: Adjust deprecation message, use `info` instead of `warning`

* plotting: Fix typo

* refactor: Rename `filename` to `file_path`

Fits better and does avoid shadowing with filename from outer scope.

* chia|tests: Move some methods from `plotting.manager` to `plotting.util`

* plotting: Make `refresh_callback` mandatory

arvidn committed:
test agg sig conditions (#7962)

* agg sig conditions always take two args

* extend test a bit, add comment

* unit test agg sig parser where args list is not null terminated

* fix formatting

dependabot[bot] committed:
Bump watchdog from 2.1.2 to 2.1.3 (#8017)

dependabot[bot] committed:
Bump actions/setup-node from 2.3.0 to 2.4.0 (#8016)

dependabot[bot] committed:
Bump github/super-linter from 4.5.1 to 4.6.1 (#8015)

Bumps [github/super-linter](https://github.com/github/super-linter) from 4.5.1 to 4.6.1.
- [Release notes](https://github.com/github/super-linter/releases)
- [Commits](https://github.com/github/super-linter/compare/v4.5.1...v4.6.1)

---
updated-dependencies:
- dependency-name: github/super-linter
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

rostislav committed:
Add test of pairing cache for BLS signature verification (#8028)

Co-authored-by: Rostislav <rostislav@users.noreply.github.com>

Quexington committed:
Added some new class methods to the Program objects (#8041)

paninaro committed:
Create SSL keys/certs with proper permissions, also verify/fix existing installs (#7972)

* Create directories with 755 permissions, SSL certs with 644, and keys with 600.

* Check SSL file permissions during chia_init(). Exits if permissions are incorrect.

* Overwrite certs/keys instead of failing to write

* Skip SSL file permission checks on Windows (requires checking ACLs)

* Check SSL file permissions when creating an ssl_context

* Skip check_ssl on Windows.
Handle some SSLInvalidPermissions exceptions.

* Added a few comments

* Added chia init --fix-ssl-permissions option to attempt to fix SSL file permission issues.
Update imported cert permissions when using chia init -c.
Code cleanup/restructuring.

* Return a tuple instead of a list

* LGTM and other minor fixes

* Fixed SSL test breakage when calling ssl_context_for_client. The ca_crt param was being passed in as both the cert and private key, triggering the permission check failure.

* Don't exit if SSL file permissions issues are found

* Tweak the exception types that are raised from traverse_dict

emlowe committed:
Manually update version in package.json (#8057)

fchirica committed:
Enable querying AAAA records for DNS Introducer. (#8008)

* Initial commit DNS ipv6.

* Fix pyblack.

* nit.

fchirica committed:
Use ip.chia.net (#8039)

* Use ip.chia.net

* oops...

Quexington committed:
Added an RPC for coin records by multiple coin names (#7920)

Quexington committed:
Added a py.typed file in the chia directory for pep-561 compliance (#8014)

prozacchiwawa committed:
Add documentation for treehash params (#8092)

arvidn committed:
Rust conditions (#7167)

* use rust implementation of condition checker

* enable rust condition checker at specific block height

* bump switch-over height

cmmarslender committed:
Increase timeout (#8111)

* Increase timeout

* increase timeout to 30

arvidn committed:
fix mempool TX cache cost (#8054)

* fix issue where the cost of the mempool TX cache (for spend bundles that can't be included in a block yet) would not be reset when the cache was emptied

* factor out the Pending TX cache from mempool, to allow unit testing

arvidn committed:
remove commented-out debug breakpoint()s (#8114)

altendky committed:
add --paginate/--no-paginate to chia wallet get_transactions (#7977)

* add --paginate/--no-paginate to chia wallet get_transactions

The default remains to paginate if sys.stdout.isatty(), otherwise it
defaults to not paginating.  This addresses cases such as piping and
output redirection to a file where the command previously just hung
while waiting for the user to press c for the next page.

* shorten lines for linting

* avoid BrokenPipeError such as with head

* stop using click's undocumented PacifyFlushWrapper

emlowe committed:
New call for logging peer_host (#8108)

almogdepaz committed:
Sync faster (#7665)

* better concurrency for sync_from_fork_point

* dedup code in wallet

* revert

* lint

* lint

* logs

* lint

* lint

* error handling, async bug fix

* lint

* break fetch if bath failed coroutine

* remove outdated comment

* consolidate duplicate code

* consolidate duplicate code

* notify wallet after each batch

* typo

* add try/catch/finally

* typing, check stale peer

* comment on worker shutdown

arvidn committed:
tolerate fields replacing network constants in config.yaml that don't exist, but print warning (#8121)

emlowe committed:
Check AttributeError for non-Optional members during node shutdown (#8109)

* Check AttributeError for non-Optional elements

* Changed to use hasattr

wjblanke committed:
updated gui to 80a41d8161ec9028995fec18158ac9e611b00f02

cmmarslender committed:
Clean up the GHA mac build to just build and upload to GHA (azure deals with s3 + release workflows) (#8178)

Yostra committed:
Fix single coin generator (#8185)

* fix single coin generator

* add sha256tree file

* add test for single

* pass it as array

* remove breakpoint

arvidn committed:
enable the rust condition checker unconditionally in testnet (#8122)

altendky committed:
Correct * to ** kwargs unpacking in time_out_assert() (#8160)

cmmarslender committed:
Swap back to non-chia notarize-cli (#8199)

* Swap back to non-chia notarize-cli

* Update node for mac to 14.x - newer syntax in PRs on notarize-cli require the new version

* Have cleanup script get rid of our other global node packages so that 12->14 transition works

cmmarslender committed:
Update configure testnet commands to set dns introducer (#8205)

wjblanke committed:
up request block rate limit for faster sync (#8204)

xdustinface committed:
cmds|server|util: Fix, refactor and improve SSL permission checks (#8211)

* server|util: Refactor `verify_ssl_certs_and_keys`

Let it take two lists instead of one with an tuple holding optional
paths.

* util: Introduce `get_ssl_perm_warning`

* util: Drop some redundant brackets

* util: Introduce `get_all_ssl_file_paths`

* util: Print warnings for CLI only

* cmds|util: Call `check_ssl` for all CLI commands

cmmarslender committed:
Update chiavdf to 1.0.3 (#8228)

paulhainsworth-chia committed:
changelog 1.2.4 (#8229)

* changelog 1.2.4

cmmarslender committed:
Clean up a few more potential leftover files in chia-blockchain-gui (#8230)

* Clean up a few more potential leftover files in chia-blockchain-gui

* Make shellcheck happy

cmmarslender committed:
Test install/install-gui scripts (#8241)

altendky committed:
only use the last line of chia version output in installhelper.py (#8239)

https://github.com/Chia-Network/chia-blockchain/issues/8234

The warning that is being output by `chia version` is causing confusion
when trying to parse the version.  This change ignores the warning by
only trying to parse the last line of the output as the version.  Other
changes should be considered such as not outputting the warning for
small commands like `--help` and `version`.

```
$ chia version
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
@             WARNING: UNPROTECTED SSL FILE!              @
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
Permissions 0664 for '/home/altendky/repos/chia-blockchain/mozilla-ca/cacert.pem' are too open. Expected 0644
One or more SSL files were found with permission issues.
Run `chia init --fix-ssl-permissions` to fix issues.
1.2.5.dev0
```

almogdepaz committed:
Merge branch 'weight_proofs_new_version' of github.com:Chia-Network/chia-blockchain into weight_proofs_new_version

almogdepaz committed:
add and use CompressedClassgroupElement in wp

parallel vdf validation
